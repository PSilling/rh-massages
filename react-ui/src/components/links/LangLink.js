// react imports
import React, { Component } from 'react';

// util imports
import _t from '../../util/Translations';

/**
 * Link for language switching.
 */
class LangLink extends Component {

  changeLanguage = () => {
    var locale = localStorage.getItem("sh-locale") === 'en' ? 'cs' : 'en';
    localStorage.setItem("sh-locale", locale);
    _t.setLocale(locale);
    window.location.reload();
  }

  render() {
    return(
      <button type="button" className="btn btn-link navbar-btn"
        onClick={this.changeLanguage} title={ _t.translate("Change language to czech") }>
        <span className="glyphicon glyphicon-globe"></span>&nbsp;
        { localStorage.getItem("sh-locale") === 'en' ? 'CZ' : 'EN' }
      </button>
    )
  }
}

export default LangLink
