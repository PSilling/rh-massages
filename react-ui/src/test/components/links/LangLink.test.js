// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import LangLink from '../../../components/links/LangLink';

test('renders content correctly', () => {
  const testRenderer = TestRenderer.create(<LangLink />),
        testInstance = testRenderer.root;

  let button = testInstance.findByType('button'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-globe" }),
      treeJSON = testRenderer.toJSON();

  testInstance.instance.changeLanguage = jest.fn();
  testRenderer.update(<LangLink />);
  button.props.onClick();

  expect(testInstance.instance.changeLanguage).toHaveBeenCalledTimes(1);
  expect(treeJSON).toMatchSnapshot();
});
