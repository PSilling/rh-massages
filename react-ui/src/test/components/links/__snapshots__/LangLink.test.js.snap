// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content correctly 1`] = `
<button
  className="btn btn-link navbar-btn"
  onClick={[Function]}
  title="Změnit jazyk na angličtinu"
  type="button"
>
  <span
    className="glyphicon glyphicon-globe"
  />
   
  EN
</button>
`;
